cheatsheet/main.tex

```diff
@@ -0,0 +1,346 @@
+            \begin{minted}{nix}
+"this is a string"
+''
+  a multi-line string
+''
+            \end{minted}
+        \end{minipage}
+        \\
+        % \hdashline
+        Boolean &
+        \begin{minipage}{0.5\textwidth}
+            \begin{minted}{nix}
+true, false
+            \end{minted}
+        \end{minipage}
+        \\
+        % \hdashline
+        Null &
+        \begin{minipage}{0.5\textwidth}
+            \begin{minted}{nix}
+null
+            \end{minted}
+        \end{minipage}
+        \\
+        % \hdashline
+        Integer &
+        \begin{minipage}{0.5\textwidth}
+            \begin{minted}{nix}
+123, -123
+            \end{minted}
+        \end{minipage}
+        \\
+        % \hdashline
+        Float &
+        \begin{minipage}{0.5\textwidth}
+            \begin{minted}{nix}
+3.14, -3.14
+            \end{minted}
+        \end{minipage}
+        \\
+        % \hdashline
+        Path &
+        \begin{minipage}{0.5\textwidth}
+            \begin{minted}{nix}
+/an/absolute/path
+./a/relative/path
+            \end{minted}
+        \end{minipage}
+        \\
+        % \hdashline
+        Attribute Set &
+        \begin{minipage}{0.5\textwidth}
+            \begin{minted}{nix}
+{ a = 1; b = 2; }
+            \end{minted}
+        \end{minipage}
+        \\
+        % \hdashline
+        List &
+        \begin{minipage}{0.5\textwidth}
+            \begin{minted}{nix}
+[ 1 2.0 "thing" null ]
+            \end{minted}
+        \end{minipage}
+        \\
+        % \hdashline
+        Comment &
+        \begin{minipage}{0.5\textwidth}
+            \begin{minted}{nix}
+# A single line comment.
+/*
+  A multi-line comment.
+*/
+            \end{minted}
+        \end{minipage}
+    \end{tabular}
+\end{ctexpression}
+
+\vfill
+
+\begin{ctexpression}{Scoping}
+{\color{nixdarkblue}Define local variables using} let...in
+    \begin{minted}{nix}
+let
+  x = 1;
+  y = 2;
+in
+  x + y # ⇒ returns 3
+    \end{minted}
+    \tcbline
+{\color{nixdarkblue}Add variables into scope using} inherit
+    \begin{minted}{nix}
+let x = 1; in { inherit x; }
+    \end{minted}
+{\color{nixdarkblue}desugars to}
+    \begin{minted}{nix}
+let x = 1; in { x = x; }
+    \end{minted}
+    \tcbline
+{\color{nixdarkblue}Add attributes into scope using} inherit
+    \begin{minted}{nix}
+let x = { y = 1; }; in { inherit (x) y; }
+    \end{minted}
+{\color{nixdarkblue}desugars to}
+    \begin{minted}{nix}
+let x = { y = 1; }; in { y = x.y; }
+    \end{minted}
+    \tcbline
+{\color{nixdarkblue}Add all attributes into scope using} with
+    \begin{minted}{nix}
+let
+  set = { x = 1; y = 2; };
+in
+  with set;
+  x + y # ⇒ returns 3
+    \end{minted}
+\end{ctexpression}
+
+\begin{ctexpression}{Conditionals}
+{\color{nixdarkblue}Define conditionals using} if...then...else
+    \begin{minted}{nix}
+if x > 0
+then 1
+else -1
+    \end{minted}
+\end{ctexpression}
+
+\vfill
+
+\begin{ctexpression}{Attribute Sets}
+{\color{nixdarkblue}Update attribute sets using} //
```

Comment:
```suggestion
{\color{nixdarkblue}Union attribute sets using} //
```
Set is not being updated, attributes of these sets are being added together.
Mathematical word from set theory for that is union. I would say it's union with right preference.

Reply:
I noticed that in nix manual it does say update
you can mark this conversation as resolved